### DISCUSSION TITLE
Auto-settings rule based on "category"

### DISCUSSION THREAD

**Discussion by thecheesemo** [gildas-lormeau/SingleFile#986] [Ideas]
I have been using SingleFile extensively to manually save pages for later reference. I do this mainly for archiving purposes and because i have a bad memory. I've since realized that I always want to save every news article I read. So, I've started to define Auto-settings rules with a specific SingleFile profile for each url of news sites i visit, This profile auto saves pages, and that works fine. However, like most people on the web now, i don't go to just one news source, I get most of my news from link aggregators like google news and reddit. I have to manually add each news website's top level domain (TLD) to the Auto-settings rule with my news autosave profile. 

**Could it be possible to create an option in the Auto-settings that would let you define a profile based on the "category" of the TDL?** 

I don't even know if there exists such a list that can be referenced, or how one would go about making one other then a manual process. But if such a list does exist, or can be created easily enough, that would be a pretty great feature.

**Comment by gildas-lormeau** [comment]
That's an interesting and inspiring feature request. Depending on how you want to categorize pages, this might be possible by extracting the schema.org (see https://schema.org) metadata from the page. 

For example, https://edition.cnn.com/2022/06/26/europe/russia-invasion-ukraine-06-26-intl/index.html contains the following HTML content.
```html
<article ... itemtype="https://schema.org/NewsArticle">...</article>
````

On https://www.nytimes.com/2022/06/26/health/abortion-medication-pills.html, the following LD+JSON script can be found.
```html 
<script ... type="application/ld+json">{ ... "@type":"NewsArticle" ... }</script>
```

Both pages define a `NewsArticle` type of metadata.

Similarly, they also include an Open Graph (see https://ogp.me) tag.
```html
<meta property="og:type" content="article">
```

So, we could imagine that SingleFile would extract all the types found on a page and offer a way to trigger the auto-save depending on these values.

For example, the condition of the rule could be `schema.type:NewsArticle` or `og.type:article`. Would that suit your needs?

**Comment by thecheesemo** [comment]
That seems like it would work! Those details are a bit about my head, but you described it well, and i understand. If I understand correctly, those are all voluntary and I imagine not every site will include any of those, but it sounds like the right way to do it.

**Reply by gildas-lormeau** [reply]
Indeed, this is all voluntary. However, news websites include most of the time these data because it helps them to appear in a friendly way on websites like Google News, facebook, Twitter etc.
I don't hide you that it's a bit of work to implement this feature, and it may be implemented in several steps. It will take me some time to think about the roadmap and to develop it but I like the idea :).
